packages/trace-viewer/src/ui/workbenchLoader.tsx

```diff
@@ -23,50 +23,48 @@ import { DialogToolbarButton } from '@web/components/dialogToolbarButton';
 import { Dialog } from '@web/shared/dialog';
 import { DefaultSettingsView } from './defaultSettingsView';
 import { TraceModelContext } from './traceModelContext';
+import { getMimeTypeForPath } from '@isomorphic/mimeType';
 
 export const WorkbenchLoader: React.FunctionComponent<{
 }> = () => {
   const [isServer, setIsServer] = React.useState<boolean>(false);
   const [traceURL, setTraceURL] = React.useState<string>();
-  const [uploadedTraceName, setUploadedTraceName] = React.useState<string>();
   const [model, setModel] = React.useState<MultiTraceModel>(emptyModel);
   const [progress, setProgress] = React.useState<{ done: number, total: number }>({ done: 0, total: 0 });
   const [dragOver, setDragOver] = React.useState<boolean>(false);
   const [processingErrorMessage, setProcessingErrorMessage] = React.useState<string | null>(null);
   const [fileForLocalModeError, setFileForLocalModeError] = React.useState<string | null>(null);
   const [showProgressDialog, setShowProgressDialog] = React.useState<boolean>(false);
 
-  const processTraceFiles = React.useCallback((files: FileList) => {
-    const url = new URL(window.location.href);
-    if (!files.length)
+  const processTraceFiles = React.useCallback((files?: FileList | null) => {
+    if (!files?.length)
       return;
-    const file = files.item(0)!;
+
+    // Do best effort to select the first valid trace valid, if not, rely on error reporting of sw
```

Comment:
Pasting multiple files looks like a user error, we can bail sooner or leave it as is.